src/plugins/upf/upf_api.c

```diff
@@ -136,11 +147,108 @@ static void vl_api_upf_app_flow_timeout_set_t_handler
   vl_api_upf_app_flow_timeout_set_reply_t *rmp = NULL;
   upf_main_t *sm = &upf_main;
 
-  //rv = upf_flow_timeout_update(mp->type, mp->default_value);
+  //rv = upf_flow_timeout_update(mp->type, ntohs(mp->default_value));
 
   REPLY_MACRO (VL_API_UPF_APP_FLOW_TIMEOUT_SET_REPLY);
 }
 
+static void
+send_upf_applications_details (vl_api_registration_t * reg,
+                               u8 * app_name, u32 flags, u32 context)
+{
+  vl_api_upf_applications_details_t *mp;
+  upf_main_t *sm = &upf_main;
+  u32 name_len;
+
+  mp = vl_msg_api_alloc (sizeof (*mp));
+  clib_memset (mp, 0, sizeof (*mp));
+
+  mp->_vl_msg_id = htons (VL_API_UPF_APPLICATIONS_DETAILS
+			  + sm->msg_id_base);
+  mp->context = context;
+
+  name_len = clib_min(vec_len(app_name) + 1, ARRAY_LEN(mp->name));
+  memcpy (mp->name, app_name, name_len-1);
```

Comment:
you need to alloc mp->name

this should also work:
```
vec_validate_init_c_string(mp->name, app_name, vec_len(app_name));
```

Reply:
Well, `mp->name` is not a vector, it's just an array in the struct, as defined here

```
define upf_applications_details {
  u32 context;

  u8 name[64];
  u32 flags;
};
```

Reply:
Here's similar piece of code in another plugin:
https://github.com/travelping/vpp/blob/feature/1908/upf/src/plugins/memif/memif_api.c#L329-L330

Reply:
ah, right, didn't see that